core/src/types/config.rs

```diff
@@ -44,14 +44,82 @@ pub struct AWSConfig {
     pub artifact_bucket_name: String,
 }
 
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct AppConfig {
+pub type SharedConfigRef = Arc<RwLock<SharedConfig>>;
+
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+pub struct SharedConfig {
     #[serde(default, rename = "repoPath", alias = "repo_path")]
     pub repo_path: String,
 
     #[serde(default, rename = "repoUrl", alias = "repo_url")]
     pub repo_url: String,
 
+    #[serde(default, rename = "githubPAT", skip_serializing_if = "Option::is_none")]
+    pub github_pat: Option<String>,
+
+    #[serde(default)]
+    pub initialized: bool,
+}
+
+impl SharedConfig {
+    pub fn initialize_repo_config(&self) -> Result<RepoConfig> {
+        if self.repo_path.is_empty() {
+            return Ok(Default::default());
+        }
+
+        // todo: pass yaml file path as a parameter
+        let config_file = PathBuf::from(self.repo_path.clone()).join("friendshipper.yaml");
+
+        if !config_file.exists() {
+            return Ok(Default::default());
+        }
+
+        let settings = Config::builder()
+            .add_source(config::File::with_name(config_file.to_str().unwrap()))
+            .set_default("trunkBranch", "main")?
+            .set_default("useConventionalCommits", false)?
+            .build()?;
+
+        // TODO: We'll need some better error handling here. Because the config file is stored
+        // in the project repo itself, all users are impacted by bad config being committed to it.
+        settings
+            .try_deserialize::<RepoConfig>()
+            .map_err(|e| anyhow!(e))
+    }
+
+    pub fn ensure_github_pat(&self) -> Result<String, CoreError> {
+        match self.github_pat.clone() {
+            Some(pat) => Ok(pat),
+            None => Err(anyhow!(
+                "GitHub PAT is not configured. Please configure it in the settings.".to_string()
+            )
+            .into()),
+        }
+    }
+}
+
+pub type BirdieConfigRef = Arc<RwLock<BirdieConfig>>;
+
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+pub struct BirdieConfig {
```

Comment:
wdyt about moving `BirdieConfig` out of `core`, renaming `AppConfig` to `FriendshipperConfig`, and moving that out of `core` too?

Reply:
that is the plan, I wanna do this is a second PR
becuase there's just so much going on

Reply:
totally fine to punt on that for this PR then IMO